### ISSUE TITLE
flat-field failing for NIRSpec lamp exposures

### ISSUE THREAD

**Issue by stscijgbot-jp** [spacetelescope/jwst#5355]
_Issue [JP-1723](https://jira.stsci.edu/browse/JP-1723) was created on JIRA by [Howard Bushouse](https://jira.stsci.edu/secure/ViewProfile.jspa?name=bushouse):_

The flat_field step is mistakenly treating NRS_AUTOWAVE lamp exposures as an imaging type, which immediately leads to failure, because there's no FLAT reference file available. This is due to the NRS_AUTOWAVE exp_type not being included in the list of modes to be handled as spectroscopic within the step.

**Comment by stscijgbot-jp** [comment]
_Comment by [Howard Bushouse](https://jira.stsci.edu/secure/ViewProfile.jspa?name=bushouse) on [JIRA](https://jira.stsci.edu/browse/JP-1723?focusedCommentId=485988#comment-485988):_

Another error signature that occurs when OPMODE='IFU' is:
```java
  File "/Users/bushouse/JWST/softw/jwst/jwst/flatfield/flat_field_step.py", line 122, in process
    output_model, flat_applied = flat_field.do_correction(
  File "/Users/bushouse/JWST/softw/jwst/jwst/flatfield/flat_field.py", line 78, in do_correction
    flat_applied = do_nirspec_flat_field(output_model, fflat, sflat, dflat,
  File "/Users/bushouse/JWST/softw/jwst/jwst/flatfield/flat_field.py", line 293, in do_nirspec_flat_field
    raise RuntimeError(f'No flat field algorithm exists for handling data {output_model}')
RuntimeError: No flat field algorithm exists for handling data <ImageModel(2048, 2048) from jw00626030001_02103_00005_nrs1_rate.fits>
```
which appears to be due to this block of code:
```java
if not hasattr(output_model, "slits"):
    if exposure_type == "NRS_IFU":
        if not isinstance(output_model, datamodels.IFUImageModel):
            log.error("NIRSpec IFU data is not an IFUImageModel; "
                      "don't know how to process it.")
            raise RuntimeError("Input is {}; expected IFUImageModel".format(type(output_model)))
        return nirspec_ifu(output_model, f_flat_model, s_flat_model, d_flat_model, dispaxis,
                           user_supplied_flat=user_supplied_flat, inverse=inverse)
    else:
        raise RuntimeError(f'No flat field algorithm exists for handling data {output_model}')
```
It looks like the "if exposure_type == "NRS_IFU" check might simply need to be updated to also include "NRS_AUTOWAVE".

**Comment by stscijgbot-jp** [comment]
_Comment by [Howard Bushouse](https://jira.stsci.edu/secure/ViewProfile.jspa?name=bushouse) on [JIRA](https://jira.stsci.edu/browse/JP-1723?focusedCommentId=486240#comment-486240):_

Fixed by #5356
